perses/app/fah_generator.py

```diff
@@ -0,0 +1,348 @@
+                             'lambda_sterics_delete': f"select(step({x} - 0.5), 2.0 * ({x} - 0.5), 0.0)",
+                             'lambda_electrostatics_insert': f"select(step({x} - 0.5), 2.0 * ({x} - 0.5), 0.0)",
+                             'lambda_electrostatics_delete': f"select(step({x} - 0.5), 1.0, 2.0 * {x})",
+                             'lambda_bonds': x,
+                             'lambda_angles': x,
+                             'lambda_torsions': x}
+
+
+def make_neq_integrator(nsteps_eq=250000, nsteps_neq=250000, neq_splitting='V R H O R V', timestep=4.0 * unit.femtosecond, alchemical_functions = DEFAULT_ALCHEMICAL_FUNCTIONS, **kwargs):
+    """
+    generate an openmmtools.integrators.PeriodicNonequilibriumIntegrator
+
+    arguments
+        nsteps_eq : int, default=1000
+            Number of equilibration steps to dwell within lambda = 0 or 1 when reached
+        nsteps_neq : int, default=1000
+            Number of nonequilibrium switching steps for 0->1 and 1->0 switches
+        neq_splitting : str, default='V R H O R V'
+            Sequence of "R", "V", "O" (and optionally "{", "}", "V0", "V1", ...) substeps to be executed each timestep.
+            "H" increments the global parameter `lambda` by 1/nsteps_neq for each step and accumulates protocol work.
+        timestep : int, default=4.0 * unit.femtosecond
+            integrator timestep
+        **kwargs :
+            miscellaneous arguments for openmmtools.integrators.LangevinIntegrator
+
+    returns
+        integrator : openmmtools.integrators.PeriodicNonequilibriumIntegrator
+
+    """
+    from openmmtools.integrators import PeriodicNonequilibriumIntegrator
+    #from copy import deepcopy
+    #integrator_kwargs = deepcopy(setup_options_dict)
+    integrator = PeriodicNonequilibriumIntegrator(alchemical_functions, nsteps_eq, nsteps_neq, neq_splitting,timestep=timestep)
+    return integrator
+
+def relax_structure(temperature, system, positions, nequil = 4, n_steps_per_iteration=250):
+    """
+    arguments
+        temperature : simtk.unit.Quantity with units compatible with kelvin
+            temperature of simulation
+        system : openmm.System
+            system object for simulation
+        positions : simtk.unit.Quantity of shape (natoms,3) with units compatible with nanometers
+            Positions of the atoms in the system
+        nequil : int
+            number of equilibration applications
+        n_steps_per_iteration : int
+            numper of steps per nequil
+
+    return
+        state : openmm.State
+            state of simulation (getEnergy=True, getForces=True, getPositions=True, getVelocities=True, getParameters=True)
+    """
+
+    from perses.dispersed.feptasks import minimize
+    from openmmtools.integrators import LangevinIntegrator
+    import tqdm
+    _logger.info(f'Starting to relax')
+    integrator = LangevinIntegrator(temperature = temperature)
+    context = openmm.Context(system, integrator)
```

Comment:
@hannahbrucemacdonald : You're using the CUDA platform here without setting `Precision` to `mixed` and the `DeeterministicForce` to `true`! This may be responsible for the discrepancies we're seeing.

Suggest you change this to use an explicit `platform_name`:
```
platform = openmm.Platform.getPlatformByName(platform_name)
if platform_name in ['CUDA', 'OpenCL']:
    platform.setPropertyDefaultValue('Precision', 'mixed')
if platform_name in ['CUDA']:
    platform.setPropertyDefaultValue('DeterministicForces', 'true')
context = openmm.Context(system, integrator, platform)
```